### PR TITLE
Add cross-origin isolation flag

### PR DESCRIPTION
**Pull request by SamVerschueren**
This PR adds a new `crossOriginIsolated` flag which in turn will set the `corp=1` query param when loading the embedded project.

It will also add the `cross-origin-isolated` flag to the allow list of the iframe if it's not yet there.

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-	"printWidth": 100,
-	"singleQuote": true,
-	"useTabs": false,
-	"tabWidth": 2,
-	"endOfLine": "lf",
-	"semi": true,
-	"arrowParens": "always",
-	"bracketSpacing": true,
-	"trailingComma": "es5"
+  "printWidth": 100,
```
Comment (SamVerschueren) [inline .prettierrc:2]: I ran prettier and it fixed some stuff.

File: examples/index.html [hunk 2]
```diff
@@ -19,7 +19,8 @@
   <body>
     <h1>StackBlitz SDK Examples</h1>
     <ul>
-      <li><a href="/examples/open-embed-project-id/">Open and embed a StackBlitz project</a></li>
+      <li><a href="/examples/open-embed-project-id/">Open and embed a StackBlitz EngineBlock project</a></li>
+      <li><a href="/examples/open-embed-webcontainer/">Open and embed a StackBlitz WebContainer project</a></li>
```
Comment (SamVerschueren) [inline examples/index.html:23]: Added a new example to test loading WebContainer projects with and without the CORP headers.

File: examples/open-embed-webcontainer/index.html [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width" />
+    <script type="module" src="./index.ts"></script>
+  </head>
+  <body>
+    <header>
+      <h1>Open and embed a StackBlitz WebContainer project</h1>
+      <div>
+        <label>
+          <input type="checkbox" name="corp" /> Cross-Origin Isolation
```
Comment (SamVerschueren) [inline examples/open-embed-webcontainer/index.html:14]: Checking and unchecking this checkbox will load the page either with or without the CORP headers making it possible to test how embeds behave in different scenario's.

File: src/helpers.ts [hunk 4]
```diff
@@ -81,3 +82,19 @@ function setFrameDimensions(frame: HTMLIFrameElement, options: EmbedOptions = {}
     frame.setAttribute('style', 'width:100%;');
   }
 }
+
+function setFrameAllowList(
+  target: HTMLElement & { allow?: string },
+  frame: HTMLIFrameElement,
+  options: EmbedOptions = {}
+) {
+  const allowList = target.allow?.split(';')?.map((key) => key.trim()) ?? [];
```
Comment (SamVerschueren) [inline src/helpers.ts:91]: We copy the allow list which is already part of the iframe, and add `cross-origin-isolated` if it's not yet there.

**Review by Nemikolh — APPROVED** [review]
Looks good! :tada:

File: CHANGELOG.md
[the remark this answers is not in the document]
Comment (SamVerschueren) [reply CHANGELOG.md:1]: I don't like doing that in the same PR. I'll create a separate commit for the release.

[the remark this answers is not in the document]
Comment (Nemikolh) [reply CHANGELOG.md:1]: Ah ok no problem! :+1:

**Pull request merged**